c7n/filters/core.py

```diff
@@ -346,11 +346,12 @@ def __call__(self, r):
 
     def process_set(self, resources, event):
         rtype_id = self.get_resource_type_id()
-        resource_map = {r[rtype_id]: r for r in resources}
+        compiled = jmespath.compile(rtype_id)
```

Comment:
why is this needed?

Reply:
for k8s resources, the unique identifier is `metadata.uid` so we can't just look at a top level key